zookeeper-server/src/test/java/org/apache/zookeeper/server/FinalRequestProcessorTest.java

```diff
@@ -227,4 +232,40 @@ private void assertMasked(boolean masked) {
         assertThat("Original ACL list has been modified", testACLs.get(2).getId().getScheme(), equalTo("world"));
         assertThat("Original ACL list has been modified", testACLs.get(2).getId().getId(), equalTo("anyone"));
     }
+
+    @Test
+    public void testGetChildrenList_Empty_List() throws IOException {
+        GetChildrenListRequest getChildrenListRequest = new GetChildrenListRequest();
+        getChildrenListRequest.setPathList(new ArrayList<>());
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        BinaryOutputArchive boa = BinaryOutputArchive.getArchive(baos);
+        getChildrenListRequest.serialize(boa, "request");
+        baos.close();
+
+        Request r = new Request(cnxn, 0, 0, ZooDefs.OpCode.getChildrenList,
+                                ByteBuffer.wrap(baos.toByteArray()), new ArrayList<Id>());
+        processor.processRequest(r);
+
+        GetChildrenListResponse rsp = (GetChildrenListResponse)responseRecord[0];
+        assertTrue(rsp.getChildren().isEmpty());
+    }
+
+    @Test
+    public void testGetChildrenList_Normal() throws IOException {
+        GetChildrenListRequest getChildrenListRequest = new GetChildrenListRequest();
+        getChildrenListRequest.setPathList(Arrays.asList("/testPath", "/testPath"));
```

Comment:
Why do you search for the same path twice?
Is it a test case to validate the results won't be merged?